### PR TITLE
feat: show/hide stack count in path segment

### PR DESCRIPTION
**Pull request by lnu**
### Prerequisites

- [x] I have read and understand the `CONTRIBUTING` guide
- [x] The commit message follows the [conventional commits][cc] guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

### Description

#634 
Adds the stack count in path segment when using pushd/popd

~~WIP. needs to make it work in Fish.~~

There is room for improvement(maybe with a template) but it works as requested for now.

[cc]: https://www.conventionalcommits.org/en/v1.0.0/#summary

### CODE REVIEW

File: src/init/omp.fish [hunk 1]
```diff
@@ -1,6 +1,7 @@
 set -g posh_theme ::CONFIG::
 
 function fish_prompt
+    set stack_count 0 #(dirs -p | wc -l)
```
Comment (JanDeDobbeleer) [inline src/init/omp.fish]: why not local?

File: src/main.go [hunk 2]
```diff
@@ -136,6 +137,10 @@ func main() {
 			"rprompt-offset",
 			40,
 			"Offset the right prompt with x when using --export-img"),
+		StackCount: flag.Int(
+			"stack-count",
+			0,
+			"The current stack count"),
```
Comment (JanDeDobbeleer) [inline src/main.go]: I would reword this to `The current location stack count`

File: src/segment_path.go [hunk 3]
```diff
@@ -41,6 +41,8 @@ const (
 	MappedLocations Property = "mapped_locations"
 	// MappedLocationsEnabled enables overriding certain locations with an icon
 	MappedLocationsEnabled Property = "mapped_locations_enabled"
+	// StackCountVisible enables the stack count display
+	StackCountVisible Property = "stack_count_visible"
```
Comment (JanDeDobbeleer) [inline src/segment_path.go]: In line with other properties I would rename this with `stack_count_enabled`

File: src/init/omp.fish (hunk 1 above)
Comment (lnu) [reply src/init/omp.fish]: I was quickly testing and I forgot to add `- l`. It should count correctly on all supported shells now.

File: src/segment_path.go (hunk 3 above)
Comment (lnu) [reply src/segment_path.go]: Yep, i put enabled at first. Will set it back.

**Comment by lnu** [comment]
still need to add some tests
